### PR TITLE
PIM-4756: Fixed string filter that starts with '0'

### PR DESCRIPTION
**Pull request by ClementGautier**
| Q                  | A
| ------------------ | ---
| Migration script?  | N
| Behats             | Y
| Specs & Static     | Y
| Changelog updated  | Y

### CODE REVIEW

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php [hunk 1]
```diff
@@ -59,22 +59,25 @@ public function apply(FilterDatasourceAdapterInterface $ds, $data)
...
+        if (FilterType::TYPE_EMPTY === $data['type']) {
+            $data['value'] = '';
+
+            return $data;
+        }
+
         $data['value'] = preg_quote($data['value']);
 
-        $data['type']  = isset($data['type']) ? $data['type'] : null;
+        if (null === $data['value'] || strlen($data['value']) === 0) {
```
Comment (skeleton) [inline src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php]: yoda

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php [hunk 2]
```diff
@@ -59,22 +59,25 @@ public function apply(FilterDatasourceAdapterInterface $ds, $data)
...
 
-        if ('in' === $data['type']) {
-            $data['value'] = explode(',', $data['value']);
+            return false;
```
Comment (skeleton) [inline src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php]: please remove blank line before `return`

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ## Bug fixes
 - PIM-4706: Product association import error with identifier containing comma or semicolon
+- PIM-4756: Filter issue with the option "start with"  "0" on the product grid
```
Comment (skeleton) [inline CHANGELOG.md]: remove extra space between "start with"  and "0"

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php [hunk 4]
```diff
@@ -59,22 +59,25 @@ public function apply(FilterDatasourceAdapterInterface $ds, $data)
...
         }
 
-        if ('empty' === $data['type']) {
-            $data['value'] = '';
+        if ('in' === $data['type']) {
```
Comment (jjanvier) [inline src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php]: no const for that ?

**Comment by jjanvier** [comment]
fix tiny comments then GTM

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php (hunk 1 above)
Comment (phaseinducer) [inline src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php]: why not `'' === $data['value']` ?

File: features/product/filtering/filter_products.feature [hunk 5]
```diff
@@ -20,14 +20,15 @@ Feature: Filter products
...
       | ebook  |           | yes     | eBook       | Ebook        | My ecommerce ebook info | My mobile ebook info | Ma info ebook ecommerce  | Ma info ebook mobile  |                  |                  |
       | chair  | furniture | yes     | Chair/Slash | Chaise/Slash | My ecommerce chair .    | My mobile chaise     | Ma info chaise ecommerce | Ma info chaise mobile |                  |                  |
     And I am logged in as "Mary"
 
   Scenario: Successfully filter products
     Given I am on the products page
-    Then the grid should contain 5 elements
-    And I should see products postit and book and book2 and ebook and chair
+    Then the grid should contain 6 elements
+    And I should see products postit and book and book2 and ebook and chair and 01234
```
Comment (sumbobyboys) [inline features/product/filtering/filter_products.feature]: `And I should see products postit, book, book2, ebook, chair and 01234` should work :)

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php (hunk 2 above)
Comment (ClementGautier) [inline src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php]: it's defined as a rule of the Symfony coding standards : http://symfony.com/doc/current/contributing/code/standards.html 

You have to put a blank line before return statements unless the return is alone inside a statement-group

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php (hunk 4 above)
Comment (ClementGautier) [inline src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php]: I just moved this part but yes I think I can make it use the constant ;)

File: src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php (hunk 2 above)
Comment (skeleton) [inline src/Pim/Bundle/FilterBundle/Filter/ProductValue/StringFilter.php]: But it's the case, no ? `return false;`is alone inside `if (null === $data['value'] || strlen($data['value']) === 0) {`

**Comment by skeleton** [comment]
GTM

**Pull request merged**